### PR TITLE
Updates to CI and fix tests

### PR DESCRIPTION
**Pull request by naveen521kk**
- tests: fix test warning, use absolute path
- Don't use `windows-2016` (it's nearing EOL) see #66 
- Upgrade dependencies for the build
- Don't build subprojects manually on macOS build
- Add separate test workflow and not use docker in tests
- Only upload wheels if tests pass
- Include license files in all the wheels
- Link wheels statically on macOS

### CODE REVIEW

**Review by PhilippImhof — APPROVED** [review]
LGTM. Do we have to do something about the failing build?

```
cibuildwheel 2.x no longer supports Python < 3.6. Please use the 1.x series or update CIBW_SKIP
Error: Command source packing/build_pango_mac.sh && pip install cython && cd manimpango && cythonize cmanimpango.pyx -3 -k -f && cd ../ && pip install . && pkg-config --libs pango failed with code 1. None

Error: Process completed with exit code 1.
```

**Comment by naveen521kk** [comment]
> Do we have to do something about the failing build?

Ah, sorry I didn't mark it as a draft. It's still a WIP.

**Comment by PhilippImhof** [comment]
> Ah, sorry I didn't mark it as a draft. It's still a WIP.

OK, fine. You actually seemed to even have requested a review :)

**Comment by naveen521kk** [comment]
> You actually seemed to even have requested a review :)

I guess that's because you are a Code owner here,

**Comment by naveen521kk** [comment]
@PhilippImhof This PR is ready and waiting for your review :).

File: tests/test_fonts.py [hunk 1]
```diff
@@ -83,7 +83,10 @@ def test_warning(font_file, font_name):
...
     )
     a = subprocess.run(
-        [sys.executable, "-c", command], check=True, stderr=subprocess.PIPE
+        [sys.executable, "-c", command],
+        check=True,
+        stderr=subprocess.PIPE,
+        cwd=Path(__file__).parent.parent,
     )
     captured = a.stderr.decode()
     assert "Pango-WARNING **" not in captured, "Looks like pango raised a warning?"
```
Comment (PhilippImhof) [inline tests/test_fonts.py:92]:
```suggestion
    assert "Pango-WARNING **" not in captured, "Looks like Pango raised a warning?"
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -55,7 +54,7 @@ jobs:
         continue-on-error: true
         shell: bash
         run: |
-          python -m pip install cibuildwheel==2.1.3
+          python -m pip install cibuildwheel==2.3.0
           echo "$event_name"
 
       - name: Build wheels(Windows)
```
Comment (PhilippImhof) [inline .github/workflows/build.yml:60]:
```suggestion
      - name: Build wheels (Windows)
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -78,12 +77,81 @@ jobs:
         with:
           path: ./wheelhouse/*.whl
           name: wheels-${{ runner.os }}-${{ matrix.platform_id }}
+
+  test_wheels:
+    name: Test Wheels on ${{ matrix.os }}-${{ matrix.platform_id }}-${{ matrix.python-version }}
```
Comment (PhilippImhof) [inline .github/workflows/build.yml:82]:
```suggestion
    name: Test wheels on ${{ matrix.os }}-${{ matrix.platform_id }}-${{ matrix.python-version }}
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -78,12 +77,81 @@ jobs:
...
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: ${{ matrix.architecture }}
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: wheels-${{ runner.os }}-${{ matrix.platform_id }}
+          path: ~/wheelhouse
+
+      - name: Install Test Dependencies
```
Comment (PhilippImhof) [inline .github/workflows/build.yml:118]:
```suggestion
      - name: Install test dependencies
```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -78,12 +77,81 @@ jobs:
...
+        run: |
+          pip install pytest pytest-cov
+
+      - name: Install Wheels
```
Comment (PhilippImhof) [inline .github/workflows/build.yml:122]:
```suggestion
      - name: Install wheels
```

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -78,12 +77,81 @@ jobs:
...
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+
+      - uses: actions/download-artifact@v2
+        with:
+          path: downloads/
+
       - name: Publish (Release)
```
Comment (PhilippImhof) [inline .github/workflows/build.yml:146]:
```suggestion
      - name: Publish release
```

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -151,9 +219,9 @@ jobs:
           asset_content_type: application/gzip
 
   success:
-    needs: [build_wheels]
+    needs: [test_wheels]
     runs-on: ubuntu-latest
-    name: Build Wheels Sucessfully
+    name: Build and Test Wheels Successfully
```
Comment (PhilippImhof) [inline .github/workflows/build.yml:224]:
```suggestion
    name: Building and testing of wheels: success
```

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -151,9 +219,9 @@ jobs:
...
     steps:
       - name: Success
-        run: echo Build Wheels Sucessfully
+        run: echo Build and Test Wheels Successfully
```
Comment (PhilippImhof) [inline .github/workflows/build.yml:227]:
```suggestion
        run: echo "Building and testing of wheels: success"
```

File: packing/build_pango_mac.sh [hunk 9]
```diff
@@ -34,28 +24,29 @@ echo "::group::Downloading Files"
...
-python $FILE_PATH/packing/download_and_extract.py "https://mirrors.kernel.org/gnu/gperf/gperf-${GPERF_VERSION}.tar.gz" gperf
-python $FILE_PATH/packing/download_and_extract.py "https://downloads.sourceforge.net/project/libpng/libpng16/${LIBPNG_VERSION}/libpng-${LIBPNG_VERSION}.tar.xz" libpng
 python $FILE_PATH/packing/download_and_extract.py "https://github.com/harfbuzz/harfbuzz/releases/download/${HARFBUZZ_VERSION}/harfbuzz-${HARFBUZZ_VERSION}.tar.xz" harfbuzz
-python $FILE_PATH/packing/download_and_extract.py "https://zlib.net/fossils/zlib-${ZLIB_VERSION}.tar.gz" zlib
-python $FILE_PATH/packing/download_and_extract.py "https://ftp.pcre.org/pub/pcre/pcre-${PCRE_VERSION}.tar.bz2" pcre
-curl -L "https://github.com/frida/proxy-libintl/archive/0.1.tar.gz" -o 0.1.tar.gz
-tar -xf 0.1.tar.gz
-mv proxy-libintl-0.1 proxy-libintl
+
+echo "Getting Patch and Applying patch for Cairo Build"
```
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:31]:
```suggestion
echo "Fetching and applying patch for Cairo build"
```

File: packing/build_pango_mac.sh [hunk 10]
```diff
@@ -65,119 +56,87 @@ echo "::group::Removing the things from brew"
...
-      --enable-unicode-properties \
-      --enable-pcregrep-libz \
-      --enable-pcregrep-libbz2
-make
-make install
-cd ..
-echo "::endgroup::"
+export CFLAGS=" -w" # warning are just noise. Ignore it.
 
 echo "::group::Building and Install proxy-libintl"
```
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:64]:
```suggestion
echo "::group::Building and installing proxy-libintl"
```

File: packing/build_pango_mac.sh [hunk 11]
```diff
@@ -65,119 +56,87 @@ echo "::group::Removing the things from brew"
...
-
-echo "::group::Building and Installing Freetype"
-cd freetype
-./configure --without-harfbuzz --prefix=$PREFIX
-make
-make install
-cd ..
-echo "::endgroup::"
-
 echo "::group::Building and Install Fontconfig"
```
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:70]:
```suggestion
echo "::group::Building and installing Fontconfig"
```

File: packing/build_pango_mac.sh [hunk 12]
```diff
@@ -65,119 +56,87 @@ echo "::group::Removing the things from brew"
...
 echo "::endgroup::"
 
 
-echo "::group::Building and Installing Pixman"
-cd pixman
-./configure --prefix=$PREFIX
-make
-make install
-cd ..
+echo "::group::Building and Install Glib"
```
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:86]:
```suggestion
echo "::group::Building and installing Glib"
```

File: packing/build_pango_mac.sh [hunk 13]
```diff
@@ -65,119 +56,87 @@ echo "::group::Removing the things from brew"
...
+  -Dlibmount=disabled \
+  -Dtests=false \
+  --force-fallback-for=pcre,libffi,proxy-libintl,zlib \
+  --default-library=static \
+  glib_builddir glib
+meson compile -C glib_builddir
+meson install -C glib_builddir
 echo "::endgroup::"
 
 echo "::group::Building and Installing Cairo"
```
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:100]:
```suggestion
echo "::group::Building and installing Cairo"
```

File: packing/build_pango_mac.sh [hunk 14]
```diff
@@ -65,119 +56,87 @@ echo "::group::Removing the things from brew"
...
+  -Dfreetype=enabled \
+  -Dtests=disabled \
+  --force-fallback-for=expat,libpng,pixman \
+  --default-library=static \
+  cairo_builddir cairo
+meson compile -C cairo_builddir
+meson install -C cairo_builddir
 echo "::endgroup::"
 
 echo "::group::Building and Installing Harfbuzz"
```
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:114]:
```suggestion
echo "::group::Building and installing Harfbuzz"
```

File: packing/build_pango_mac.sh [hunk 15]
```diff
@@ -65,119 +56,87 @@ echo "::group::Removing the things from brew"
...
+  -Dintrospection=disabled \
+  -Dglib=disabled \
+  --default-library=static \
+  harfbuzz_builddir harfbuzz
 meson compile -C harfbuzz_builddir
 meson install -C harfbuzz_builddir
 echo "::endgroup::"
 
+
 echo "::group::Buildling and Installing Pango"
```
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:132]:
```suggestion
echo "::group::Building and installing Pango"
```

**Review by PhilippImhof — CHANGES_REQUESTED** [review]
Very well done. I have just a few nitpicks w.r.t consistent language use.

**Comment by naveen521kk** [comment]
@PhilippImhof Thanks for the review. It would be nice if you could test the macOS wheels also https://github.com/ManimCommunity/ManimPango/suites/4557070397/artifacts/122415312

**Comment by PhilippImhof** [comment]
I currently have some environment problems on my machine and thus cannot properly test the macOS wheels. But I think we can go for it, no?

**Comment by naveen521kk** [comment]
> But I think we can go for it, no?

Yeah, I guess at least it works on Github's machine.

**Review by PhilippImhof — APPROVED** [review]
LGTM

**Comment by naveen521kk** [comment]
I'll do a release later today. I guess it should be a major bump right because of https://github.com/ManimCommunity/ManimPango/pull/65

**Pull request merged**